src/api/shadow.rs

```diff
@@ -88,8 +113,14 @@ impl Shadow {
 		if resp.status() == StatusCode::INTERNAL_SERVER_ERROR {
 			Err(Error::ShadowInternalServerError(resp.text().await?).into())
 		} else {
-			let json: EthereumReceiptProofThingJson = resp.json().await?;
-			Ok(json.into())
+			let raw = resp.text().await?;
+			if raw.contains("\"error\"") {
```

Comment:
Parsing like the other two apis will cause a strange error, so here I manually check whether it is an error message.